File: exercises/src/main/java/io/miragon/training/adapter/inbound/cibseven/SendWelcomeMailDelegate.java
package io.miragon.training.adapter.inbound.cibseven;

import io.miragon.training.application.port.inbound.SendWelcomeMailUseCase;
import io.miragon.training.domain.SubscriptionId;
import org.cibseven.bpm.engine.delegate.DelegateExecution;
import org.springframework.stereotype.Component;

import java.util.UUID;

@Component
public class SendWelcomeMailDelegate extends BaseDelegate {

    private final SendWelcomeMailUseCase useCase;

    public SendWelcomeMailDelegate(SendWelcomeMailUseCase useCase) {
        this.useCase = useCase;
    }

    @Override
    protected void executeTask(DelegateExecution execution) {
        // TODO Aufgabe 1:
        //  1. Lese die Variable "subscriptionId" aus der DelegateExecution
        //     Hinweis: (String) execution.getVariable("subscriptionId")
        //  2. Erstelle eine SubscriptionId: new SubscriptionId(UUID.fromString(...))
        //  3. Rufe useCase.sendWelcomeMail(subscriptionId) auf
        throw new UnsupportedOperationException("Aufgabe 1: Verbinde den Delegate mit dem Use Case");
    }
}

File: exercises/src/main/kotlin/io/miragon/training/adapter/inbound/cibseven/SendWelcomeMailDelegate.kt
package io.miragon.training.adapter.inbound.cibseven

import io.miragon.training.application.port.inbound.SendWelcomeMailUseCase
import io.miragon.training.domain.MembershipId
import org.cibseven.bpm.engine.delegate.DelegateExecution
import org.springframework.stereotype.Component
import java.util.UUID

@Component
class SendWelcomeMailDelegate(
    private val useCase: SendWelcomeMailUseCase,
) : BaseDelegate() {

    override fun executeTask(execution: DelegateExecution) {
        // TODO Aufgabe 1:
        //  1. Lese die Variable "membershipId" aus der DelegateExecution
        //     Hinweis: execution.getVariable("membershipId") as String
        //  2. Erstelle eine MembershipId: MembershipId(UUID.fromString(...))
        //  3. Rufe useCase.sendWelcomeMail(membershipId) auf
        TODO("Aufgabe 1: Verbinde den Delegate mit dem Use Case")
    }
}
